### ISSUE TITLE
Tito complains about missing file/directory during rpm build process. (Probable cause - `copy_extra_sources`)

### ISSUE THREAD

**Issue by wolfdale** [rpm-software-management/tito#380]
This issue is observed with new version of tito (v0.6.14 & v0.6.13). 

From initial conversation with @FrostyX  - The issue is caused by `copy_extra_sources` which is introduced to deal with some
special use-cases.

Steps to reproduce -
Pre-requisite - Make sure you have tito (v0.6.14 or v0.6.13) installed.
1) ```clone candlepin project (https://github.com/candlepin/candlepin.git)```
2) ```cd candlepin/server```
3) ```tito build --test --rpm --debug --verbose --no-cleanup```

Stacktrace -- 
```
Looking for .spec.tmpl in /candlepin/server
Using file: candlepin.spec.tmpl
Command: cheetah fill --flat --pickle=/tmp/tito/rpmbuild-candlepinuXawb6/SOURCES/candlepin-git-0.fb0dd2e/tito-cheetah-pickleCZFg5y --odir=/tmp/tito/rpmbuild-candlepinuXawb6/SOURCES/candlepin-git-0.fb0dd2e --oext=cheetah /candlepin/server/candlepin.spec.tmpl
Status code: 0
Command output: Filling /candlepin/server/candlepin.spec.tmpl -> /tmp/tito/rpmbuild-candlepinuXawb6/SOURCES/candlepin-git-0.fb0dd2e/candlepin.spec.cheetah
 
Looking for .spec in /tmp/tito/rpmbuild-candlepinuXawb6/SOURCES/candlepin-git-0.fb0dd2e
Using file: candlepin.spec
Wrote: /tmp/tito/candlepin-git-0.fb0dd2e.tar.gz
Copying /tmp/tito/rpmbuild-candlepinuXawb6/SOURCES/candlepin-git-0.fb0dd2e/candlepin-3.1.15.war -> /tmp/tito/rpmbuild-candlepinuXawb6/SOURCES
WARNING: Leaving rpmbuild files in: /tmp/tito/rpmbuild-candlepinuXawb6
Traceback (most recent call last):
  File "/usr/bin/tito", line 9, in <module>
    load_entry_point('tito==0.6.14', 'console_scripts', 'tito')()
  File "/usr/lib/python2.7/site-packages/tito/cli.py", line 902, in main
    CLI().main(sys.argv[1:])
  File "/usr/lib/python2.7/site-packages/tito/cli.py", line 209, in main
    return module.main(argv)
  File "/usr/lib/python2.7/site-packages/tito/cli.py", line 384, in main
    return builder.run(self.options)
  File "/usr/lib/python2.7/site-packages/tito/builder/main.py", line 156, in run
    self.rpm()
  File "/usr/lib/python2.7/site-packages/tito/builder/main.py", line 540, in rpm
    BuilderBase.rpm(self)
  File "/usr/lib/python2.7/site-packages/tito/builder/main.py", line 263, in rpm
    self.copy_extra_sources()
  File "/usr/lib/python2.7/site-packages/tito/builder/main.py", line 512, in copy_extra_sources
    shutil.copy(src, self.rpmbuild_sourcedir)
  File "/usr/lib64/python2.7/shutil.py", line 119, in copy
    copyfile(src, dst)
  File "/usr/lib64/python2.7/shutil.py", line 82, in copyfile
    with open(src, 'rb') as fsrc:
IOError: [Errno 2] No such file or directory: '/tmp/tito/rpmbuild-candlepinuXawb6/SOURCES/candlepin-git-0.fb0dd2e/candlepin-3.1.15.war'
```

**Comment by FrostyX** [comment]
Thank you for reporting the issue @wolfdale.
We have discussed it issue off-list and you summarized the bug report perfectly, so there is nothing left to say for me except that I am going to look at it as soon as possible.

And that I will try to add some tests for `copy_extra_sources` because this is not the first bug caused by it and we should have tested that more extensively in the first place.

**Comment by FrostyX** [comment]
So I started hacking on it and realized, that this is already fixed in PR #378 by @jirihnidek.
The change just hasn't been released yet.

@wolfdale, can you please try to install Tito from `master` and test whether it works for your package? I am trying it and getting past the issue you posted but the build still fails. I doubt it is related but it would be nice if you can confirm it works for you and then I will do a new release.

**Comment by wolfdale** [comment]
@FrostyX So I have installed Tito from master and it worked perfectly with candlepin & I'm able to build rpm with it. It would be great to have new release of Tito in near future with this fix. 
Thanks @jirihnidek for the fix.

**Comment by FrostyX** [comment]
@wolfdale thank you for the confirmation, I will release today.

**Comment by FrostyX** [comment]
I have released a new version [Tito 0.6.15](https://github.com/rpm-software-management/tito/releases/tag/tito-0.6.15-1) and submitted updates for Fedora. They are still in testing but you can give karma to get them into stable sooner.

Thank you for the collaboration guys,
I am closing the issue now.
